autopush/websocket.py

```diff
@@ -45,6 +45,33 @@ class SimplePushServerProtocol(WebSocketServerProtocol):
     # Testing purposes
     parent_class = WebSocketServerProtocol
 
+    # Defer helpers
+    def deferToThread(self, func, *args, **kwargs):
+        d = deferToThread(func, *args, **kwargs)
+        self._callbacks.append(d)
+
+        def f(result):
+            if d in self._callbacks:
+                self._callbacks.remove(d)
+            return result
+        d.addBoth(f)
```

Comment:
Yea, if you look at the deferToLater I made, I call failure.Failure() inside an exception handling block, and Failure() will find it to initialize itself with the current error being handled.